### DISCUSSION TITLE
subject of email notifications inconsistent across hosts/docker installs

### DISCUSSION THREAD

**Discussion by glc650** [containrrr/watchtower#1047] [Q&A] [answered]
**Describe the bug**

I'm using watchtower on two different hosts running two different docker instances.  On one host, email notifications show the host name in the subject (Watchtower updates on <server_name>).  On the other host, I get a string of numbers/letters instead (Watchtower updates on 28959a6d25cf)

**To Reproduce**

Just wait for notification to be sent (on container/docker startup or when image updates are detected).

**Expected behavior**

Show host name in subject of email notification (like watchtower does on my other host/docker install).

**Environment**

<details>
 <summary><b>docker-compose.yml files: </b></summary>

```
The only differences are one is setup for monitor only while the other defaults to auto update with cleanup and removal of volumes unless container has the monitor only tag.
```

</details>

**Comment by piksel** [comment, marked as answer]
Watchtower uses the hostname that the environment supplies. Docker normally sets the hostname based on the UTS. My guess is that it uses `host` networking on one of the instances perhaps?

You can supply the hostname when starting a container with the `--hostname` argument (and optionally use `$(hostname)` to get the current systems hostname):
```
docker run -d \
  --hostname "$(hostname)"
  --name watchtower \
  -v /var/run/docker.sock:/var/run/docker.sock \
  containrrr/watchtower
```

Refer to the [`docker run` docs](https://docs.docker.com/engine/reference/run/#network-settings).

**Comment by glc650** [comment]
Couldn't get $(hostname) to work with compose so I just set "hostname" manually and that did the trick.
